### ISSUE TITLE
The program is not work when changing to bert-base-uncased model

### ISSUE THREAD

**Issue by ylsung** [dguo98/DiffPruning#1]
Thank you for the work.

I would like to run the DiffPrune with the bert-base model, but the model seems not to work. The following is the logs

[SST-2]
![image](https://user-images.githubusercontent.com/32589903/106550282-d8f02200-64e0-11eb-93bb-aee30c58fb07.png)
The accuracy is around `0.5`.

[CoLA]
![image](https://user-images.githubusercontent.com/32589903/106550391-0c32b100-64e1-11eb-8730-e30e0b444cb1.png)
The correlation keeps being `0`.

Because the links in `download_glue_data.py` don't work for me, I substitute them to the download links in https://gluebenchmark.com/tasks, and `run_glue_diffpruning.py` can run normally.

[Config]
I only change `model_name_or_path` to `bert-base-cased` and `BASE_DIR` to my path.

```
EXP_NAME=sst-base
SEED=0
PER_GPU_TRAIN_BATCH_SIZE=8
GRADIENT_ACC=1
LR=0.00001000
SPARSITY_PEN=0.00000012500
CONCRETE_LOWER=-1.500
CONCRETE_UPPER=1.500
ALPHA_INIT=5
FIX_LAYER=-1
BASE_DIR=.
LOCAL_DATA_DIR=${BASE_DIR}/glue_data
LOCAL_CKPT_DIR=${BASE_DIR}/logs/${EXP_NAME}
GPU=0
TASK=sst-2
DATA=SST-2

mkdir -p ${LOCAL_CKPT_DIR}

cd ${BASE_DIR}

# install transformer
CUDA_VISIBLE_DEVICES=${GPU} python ${BASE_DIR}/examples/run_glue_diffpruning.py --model_type bert --model_name_or_path bert-base-cased --task_name ${TASK} --output_dir ${LOCAL_CKPT_DIR} --do_train --do_eval --data_dir ${LOCAL_DATA_DIR}/${DATA} --sparsity_pen ${SPARSITY_PEN} --concrete_lower ${CONCRETE_LOWER} --concrete_upper ${CONCRETE_UPPER} --num_train_epochs 3 --save_steps 5000 --seed ${SEED} --per_gpu_train_batch_size ${PER_GPU_TRAIN_BATCH_SIZE} --learning_rate ${LR}  --gradient_accumulation_steps ${GRADIENT_ACC} --fix_layer ${FIX_LAYER} --max_seq_length 128 --per_gpu_eval_batch_size 8 --overwrite_output_dir --logging_steps 5000 1>${LOCAL_CKPT_DIR}/${EXP_NAME}.out 2>${LOCAL_CKPT_DIR}/${EXP_NAME}.err
```

Although the hyperparameters may not be optimal, I suppose the results should not be current numbers. Do you have experimental results on bert-base model?

**Comment by dguo98** [comment]
Hi Yi-Lin,

Can you try again with the updated codebase?  It's important to also run the other two steps. :)

After changing models, it might require more hyper-parameters tuning.
For CoLA specifically, you can try training 8 epochs of diff pruning, and use learning rate of 2e-5 and batch size of 32.
The final validation accuracy should be 57.2% and test accuracy should be 52.3%. According to TinyBERT paper, the BERT-base full fine-tuning should achieve test accuracy 52.8%.

For SST-2, you can try batch size 16 and learning rate 3e-5. The validation accuracy should be 92.1% and test accuracy 92.9%

Thanks! Let me know if you need other help :)

**Comment by dguo98** [comment]
Re-open it in case if you need other help. Thanks!

**Comment by dguo98** [comment]
Oh, I just realized that you are getting 0 and 50% accuracy. That means there's something fundamentally breaks. I just ran experiments yesterday and got the above results. That's super weird. 

Let me double check. I might have made some mistakes when copying code from my original codebase.

Meanwhile, in case it will be helpful. The correct logs for BERT-base-uncased for CoLA and SST-2 are:
SST-2: https://drive.google.com/drive/folders/153nkjNWXo2qKEcZQ5dZ9wRo4-_3gCG5S?usp=sharing
CoLA: https://drive.google.com/drive/folders/1XVkbIEIl2uzThCgapb2CFr9f1OOoOgkg?usp=sharing

I also uploaded the data I used here: https://drive.google.com/file/d/17RnJsRUZqmLArXACLxfLj62jkVEvl1ty/view?usp=sharing
